modelskill/matching.py

```diff
@@ -209,14 +214,230 @@ def _single_obs_compare(
+    for m in raw_mod_data.values():
+        if len(m) == 0:
+            continue
+        _mod_start.append(m.index[0])  # TODO: xr.Dataset
+        _mod_end.append(m.index[-1])  # TODO: xr.Dataset
+    return min(_mod_start), max(_mod_end)
+
+
+def match_time(
+    observation: Observation,
+    raw_mod_data: Dict[str, pd.DataFrame],
+    max_model_gap: Optional[TimeDeltaTypes] = None,
+) -> xr.Dataset:
+    """Match observation with one or more model results in time domain
+    and return as xr.Dataset in the format used by modelskill.Comparer
+
+    Will interpolate model results to observation time.
+
+    Note: assumes that observation and model data are already matched in space.
+
+    Parameters
+    ----------
+    observation : Observation
+        Observation to be matched
+    raw_mod_data : Dict[str, pd.DataFrame]
+        Dictionary of model results ready for interpolation
+    max_model_gap : Optional[TimeDeltaTypes], optional
+        In case of non-equidistant model results (e.g. event data),
+        max_model_gap can be given e.g. as seconds, by default None
+
+    Returns
+    -------
+    xr.Dataset
+        Matched data in the format used by modelskill.Comparer
+    """
+    _obs_name = "Observation"
+    _mod_names = list(raw_mod_data.keys())
+    _mod_start, _mod_end = _get_model_start_end(raw_mod_data)
+
+    assert isinstance(observation, (PointObservation, TrackObservation))
+    gtype = "point" if isinstance(observation, PointObservation) else "track"
+    observation = observation.copy()
+    observation.trim(_mod_start, _mod_end)
+
+    first = True
+    for name, mdata in raw_mod_data.items():
+        df = _model2obs_interp(observation, mdata, max_model_gap)
+        if gtype == "track":
+            # TODO why is it necessary to do mask here? Isn't it an error if the model data is outside the observation track?
+            df_obs = observation.data.to_pandas()  # TODO: xr.Dataset
+            _mask_model_outside_observation_track(name, df, df_obs)
+
+        if first:
+            data = df
+        else:
+            data[name] = df[name]
+
+        first = False
+
+    data.index.name = "time"
+    data = data.dropna()
+    data = data.to_xarray()
+    data.attrs["gtype"] = gtype
+
+    if gtype == "point":
+        data["x"] = observation.x
+        data["y"] = observation.y
+        data["z"] = observation.z  # type: ignore
+
+    data.attrs["name"] = observation.name
+    data.attrs["quantity_name"] = observation.quantity.name
+    data["x"].attrs["kind"] = "position"
+    data["y"].attrs["kind"] = "position"
+    data[_obs_name].attrs["kind"] = "observation"
+    data[_obs_name].attrs["unit"] = observation.quantity.unit
+    data[_obs_name].attrs["color"] = observation.color
+    data[_obs_name].attrs["weight"] = observation.weight
+    for n in _mod_names:
+        data[n].attrs["kind"] = "model"
+
+    data.attrs["modelskill_version"] = __version__
+
+    return data
+
+
+def _model2obs_interp(
```

Comment:
The name of this function doesn't reveal that it also remove gaps, also not the docstring. Should gap removal be part of this function.